models/icepack/model_mod.f90

```diff
@@ -0,0 +1,885 @@
+      base_offset = base_offset + (icat-1) * Nx
+      call lon_lat_interpolate(state_handle, ens_size, base_offset, llon, llat, set_obstype, cat_signal_interm, expected_tsfc, istatus)
+      if ((any(expected_conc<0.0) .or. any(expected_conc>1.0)) .and. (debug > 1)) then
+      print*,'obstype TSFC expected sicn:',expected_conc
+      print*,'TSFC sicn lat lon:',llat,llon
+      endif
+      if ((any(expected_tsfc>50.0) .or. any(expected_tsfc<-100.0)) .and. (debug > 1)) then
+      print*,'obstype TSFC expected tsfcn:',expected_tsfc
+      print*,'TSFC tsfcn lat lon:',llat,llon
+      endif
+      temp = temp + expected_conc * expected_tsfc  !sum(aicen*Tsfcn)
+      temp1= temp1+ expected_conc                  !sum(aicen) = aice
+   end do
+   expected_obs = temp/max(temp1,1.0e-8)  !sum(aicen*Tsfcn)/aice = Tsfc ;averaged temperature over sea-ice covered portion
+   if ((any(expected_obs>50.0) .or. any(expected_obs<-100.0)) .and. (debug > 1)) then
+      print*,'obstype TSFC expected obs:',expected_obs
+      print*,'TSFC tsfc lat lon:' ,llat,llon
+      print*,'temp:',temp
+      print*,'temp1:',temp1
+   endif
+else
+    call lon_lat_interpolate(state_handle, ens_size, base_offset, llon, llat, set_obstype, cat_signal, expected_obs, istatus)
+    
+      if (any(expected_obs<0.0) .and. (debug > 1)) then
+      print*,'obstype SIC expected concs:',expected_obs
+      print*,'SIC sic negative lat lon:',llat,llon
+      endif
+      if (any(expected_obs>1.0) .and. (debug > 1)) then
+      print*,'obstype SIC expected concs:',expected_obs
+      print*,'SIC sic positive lat lon:',llat,llon
+      endif
+endif
+
+if (cat_signal == -1) then
+      ! we need to know the aggregate sea ice concentration for these special cases
+      base_offset = get_index_start(domain_id, get_varid_from_kind(QTY_SEAICE_CONCENTR))
+      base_offset = base_offset + (cat_index-1)
+      call lon_lat_interpolate(state_handle, ens_size, base_offset, llon, llat, set_obstype, cat_signal, expected_aggr_conc, istatus)
+      expected_obs = expected_obs/max(expected_aggr_conc,1.0e-8)  ! hope this is allowed so we never divide by zero
+
+      if ((any(expected_aggr_conc<0.0) .or. any(expected_aggr_conc>1.0)) .and. (debug > 1)) then
+      print*,'obstype SIT expected conc:',expected_aggr_conc
+      print*,'SIT sic lat lon:',llat,llon
+      endif
+
+endif
+
+if (debug > 1) print *, 'interp val, istatus = ', expected_obs, istatus, size(expected_obs)
+
+! This should be the result of the interpolation of a
+! given obs_type of variable at the given location.
+
+! The return code for successful return should be 0. 
+! Any positive number is an error.
+! Negative values are reserved for use by the DART framework.
+
+end subroutine model_interpolate
+
+!------------------------------------------------------------------
+
+subroutine lon_lat_interpolate(state_handle, ens_size, offset, lon, lat, var_type, cat_signal, expected_obs, istatus)
```

Comment:
this routine is called lat_lon_interpolate, but it does not seem to interpolate.